airbyte-integrations/connectors/source-wikipedia-pageviews/source_wikipedia_pageviews/wikipedia_pageviews.yaml

```diff
@@ -0,0 +1,63 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: [ "items" ]
+  requester:
+    url_base: "https://wikimedia.org/api/rest_v1/metrics/pageviews"
+    http_method: "GET"
+    request_options_provider:
+      request_headers:
+        "User-Agent": "AirbyteWikipediaPageviewsConnector/1.0 (https://github.com/airbyte/airbyte)"
+  stream_slicer:
+    type: DatetimeStreamSlicer
+    start_datetime: "{{config.start}}"
+    end_datetime: "{{config.end}}"
+    step: "1d"
+    cursor_field: "timestamp"
+    datetime_format: "%Y%m%d"
+  per_article_requester:
+    $options:
+      $ref: "*ref(definitions.requester)"
+    path: "/per-article/{{config.project}}/{{config.access}}/{{config.agent}}/{{config.article}}/daily/{{config.start}}/{{config.end}}"
+  top_requester:
+    $options:
+      $ref: "*ref(definitions.requester)"
+    path: "/top/{{config.project}}/{{config.access}}/{{config.year}}/{{config.month}}/{{config.day}}"
```

Comment:
I don't think this is the best implementation to retrieve the data for Wikipedia. @girarda can you help here?

Reply:
I agree. A better approach would be to use a DatetimeStreamSlicer with a step="1d" and datetime_format="%Y/%m/%d".

```
stream_slicer:
  <...>
  step: "1d"
  datetime_format: "%Y/%m/%d"
```

The path can then refer to the stream slice

```
    path: "/top/{{config.project}}/{{stream_slice.start_datetime}}"
```

Reply:
> 

Hi, how would this work with two stream slicers? It seems to be using YYYYMMDD with no slashes which is what my other stream slicer should be doing.

Reply:
> I agree. A better approach would be to use a DatetimeStreamSlicer with a step="1d" and datetime_format="%Y/%m/%d".
> 
> ```
> stream_slicer:
>   <...>
>   step: "1d"
>   datetime_format: "%Y/%m/%d"
> ```
> 
> The path can then refer to the stream slice
> 
> ```
>     path: "/top/{{config.project}}/{{stream_slice.start_datetime}}"
> ```

Hi, in my config.json should I have a different start and end time for each stream slicer then? as I cannot share the start and end we have right now as that is in YYYYMMDD format. If I seperate them what are the naming conventions. Would it be `per-article-start` and `per-article-end`? or something different?

Reply:
No, you will have only one start date. Each stream can reuse the stream slicer.

Reply:
> No, you will have only one start date. Each stream can reuse the stream slicer.

One of my streams needs `%Y%m%d` and the other needs `%Y/%m/%d` how can I make this work?

Reply:
Then you need to stream slicers for each type.

Reply:
> Then you need to stream slicers for each type.

If I have a stream slicer of each type with different formats, that means I will need two different start and end requirments in the config?

Reply:
@evwltrs the datetime format for the start and end can be different from the one that is used to update the requests, so both stream slicers can use the same config field

[here's an example](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-sendgrid/source_sendgrid/sendgrid.yaml#L47-L63)

Reply:
> @evwltrs the datetime format for the start and end can be different from the one that is used to update the requests, so both stream slicers can use the same config field
> 
> [here's an example](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-sendgrid/source_sendgrid/sendgrid.yaml#L47-L63)

`"type": "CONNECTION_STATUS", "connectionStatus": {"status": "FAILED", "message": "\"Unable to connect to stream top - time data '20220101' does not match format '%Y/%m/%d'\""}}`

I assumed it didn't work like that because of this error.... any ideas?

Reply:
can you try with `%`?

`%Y%m%d`

Reply:
> can you try with `%`?
> 
> `%Y%m%d`

wow that was a classic one :D. now im having the issue again where nothing is getting appended to that part of the url.

Reply:
You have two options here:
1. add a `start_time_option` or `end_time_option` to your stream slicer
2. add a field in the requester's `request_options_provider`. its value can point to the stream slice `" {{ stream_slice['start_time'] }}"`

Reply:
For the former what options would I need to use in my situation?